pymeasure/instruments/keysight/keysightE3642A.py

```diff
@@ -0,0 +1,67 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2024 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+from pymeasure.instruments import SCPIMixin, Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+
+class KeysightE3642A(SCPIMixin, Instrument):
+    def __init__(self, adapter, name="Keysight E3642A", **kwargs):
+        """ Represents the Keysight E3642A DC Power Supply interface for interacting with
+        the instrument. """
+        super().__init__(
+            adapter, name,
+            **kwargs
+        )
+
+    output_state_enabled = Instrument.control(
+        "OUTPUT:STATE?",
+        "OUTPUT:STATE %d",
+        """Control if output state is enabled""",
```

Comment:
Docstrings should end with a period (or exclamation mark...):
```suggestion
        """Control whether output state is enabled.""",
```